### DISCUSSION TITLE
Apparent issues with MOOSE Docker Container

### DISCUSSION THREAD

**Discussion by andrewritzmann** [idaholab/moose#17812] [Q&A Getting Started] [answered]
Hi,

I was using the MOOSE docker container on both Windows 10 and Linux (Ubuntu 18.04.5 LTS) with fairly good success. Recently, I changed laptops, and I tried using the latest container of ubuntu-moose (I've tried several since, but I can confirm as of this morning that the following issue exists for tag 62256f87eb3a30538516995c7234d4a932c63f0a [current latest image as of 9:09 AM PDT on 5/12/2021]).

Following the instructions on the MOOSE Getting Started --> Docker page (https://mooseframework.inl.gov/getting_started/installation/docker.html). I ran, per the instructions, the following command:
docker run -ti herter4171/ubuntu-moose:latest /bin/bash -c 'cd test; ./run_tests'

The result was a failure due to the pyhit module. 
Message: "failed to import pyhit - try running "make hit" in the $MOOSE_DIR/test directory."
I have tried to troubleshoot this with making the hit target in both the $MOOSE_DIR/test directory and the $MOOSE_DIR/framework/contrib/hit directory. 

This has been a persistent problem of late with running the image on Windows 10. I am reluctant to try the newer container on my Linux host system for fear of messing with an older, working version. The problem appears to have occurred between March 10, 2021 and today.

Since the Docker approach is supposed to be a plug-and-play solution, fixing this issue would be quite helpful.

Best,

Andrew

**Comment by GiudGiud** [comment, marked as answer]
Hi Andrew

Using WSL is the preferred way to use MOOSE on windows 10. 
For the pyhit error, you just need to download moosetools i think. [repo](https://github.com/idaholab/moosetools)
For more[ info on pyhit](https://mooseframework.inl.gov/moose/python/pyhit/index.html)

Best,
Guillaume

**Reply by aeslaughter** [reply]
You do not need moosetools repo, something is going wrong with compiling hit python bindings. Typically, when this happens it is because of hit was compiled with one version of python and you are using a different version with run_tests.
